tests/manage/monitoring/test_workload_with_distruptions.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf8 -*-
+
+
+import logging
+
+from ocs_ci.framework import config
+from ocs_ci.framework.testlib import tier2, pre_upgrade, post_upgrade
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.ocs import fiojob
+from ocs_ci.ocs.resources.objectconfigfile import ObjectConfFile
+
+
+logger = logging.getLogger(__name__)
+
+
+"""
+Test cases test_workload_with_checksum and test_workload_with_checksum_verify
+were originaly proposed for testing metrics and (partial) cluster shutdown, but
+they can be used for any other test which needs to verify that cluster still
+preserves data written before some cluster wide distruption like shutdown,
+reboot or upgrade.
+
+The basic idea is 1st test deploys a k8s job which writes data on OCS based PV
+using fio along with a checksum file, and then the 2nd test verifies that the
+data are still there on the PV. For this to work, the PV created during the 1st
+test needs to be preserved.
+
+The data are written on a PV via workload_storageutilization_checksum_rbd
+fixture, used in test_workload_with_checksum. When the writing finishes, the
+checksum is computed and stored along with the data, and finally reclaim policy
+of the PV is changed to Retain, so that the PV survives teardwon of the test.
+The PV is labeled as ``fixture=workload_storageutilization_checksum_rbd`` so
+that the verification test can identify and reuse it.
+
+After the first test finishes, the PV is in Released state and won't be reused,
+because it contains claimRef referencing the original PVC.
+
+The test_workload_with_checksum_verify locates the PV created by
+test_workload_with_checksum via the label, and then removes it's claimRef so
+that the PV changes it's state to Available. The test asks for the PV using PVC
+with the same parameters, and executes the checksum verification.
+
+The original idea was to use fio verification feature, but testing showed that
+when this fails for some reason or gets stuck, it's hard to debug.
+"""
+
+
+@pre_upgrade
+@tier2
+def test_workload_with_checksum(workload_storageutilization_checksum_rbd):
+    """
+    Purpose of this test is to have checksum workload fixture executed.
+    """
+    msg = "fio report should be available"
+    assert workload_storageutilization_checksum_rbd['result'] is not None, msg
+    fio = workload_storageutilization_checksum_rbd['result']['fio']
+    assert len(fio['jobs']) == 1, "single fio job was executed"
+    msg = "no errors should be reported by fio when writing data"
+    assert fio['jobs'][0]['error'] == 0, msg
+
+
+@post_upgrade
+@tier2
```

Comment:
I think it would be better to have this in tier1 as we usually don't run tier2 post upgrade.